src/databricks/labs/remorph/reconcile/connectors/databricks.py

```diff
@@ -4,12 +4,18 @@
 from pyspark.sql import DataFrame
 from pyspark.sql.functions import col
 
+from databricks.labs.remorph.config import TableRecon
 from databricks.labs.remorph.reconcile.connectors.data_source import DataSource
+from databricks.labs.remorph.reconcile.constants import SourceType
 from databricks.labs.remorph.reconcile.recon_config import JdbcReaderOptions, Schema
 
 
 class DatabricksDataSource(DataSource):
-    def read_data(self, catalog: str, schema: str, query: str, options: JdbcReaderOptions) -> DataFrame:
+    def __init__(self, spark, ws, scope):
+        super().__init__(spark, ws, scope, engine=SourceType.DATABRICKS.value)
```

Comment:
not blocking this PR, but it's code-smell, because you're not injecting the instance of DatabricksDialect and use an enum instead.